### PR TITLE
Setup migrate script + DB schema

### PR DESCRIPTION
**Pull request by edisonz0718**
Add DB migration schema. Tested in dev cluster

4 tables are added:

1. allowlist: Keep track of the meta data of the current allowlists we created
2. merkle_proof: Merkle proofs for each wallet address per allow list
3. nft: Keep track of the backstory of each nft
4. chapter: Keep track of the meta data of the chapters

To simplify the design, the other schemas in the data model design are not added because:
 1. They are not needed at this point (We only use one chain and have one story). Add them create unnecessary indirection.
 2. The current tables are forward compatible, meaning that we can add the other models in the future if needed and not break the existing ones.

For example, nft table only keeps tracks of backstory but not the other traits, because the other traits are already persisted onchain. So we just need to store them in redis for fast serving.

### CODE REVIEW

File: api/migrations/000003_create_nft_table.up.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS nft (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    collection_address VARCHAR NOT NULL,
+    token_id INTEGER NOT NULL,
+    owner_address VARCHAR,
+    image_url VARCHAR,
+    traits VARCHAR,
+    backstory VARCHAR(3000),
```
Comment (LeoHChen) [inline api/migrations/000003_create_nft_table.up.sql:10]: I am thinking we may also need to save backstory on other decentralized storage, like IPFS, or Arweave. do you need to add a link to the backstory?

File: api/migrations/000004_create_chapter_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+BEGIN;
```
Comment (LeoHChen) [inline api/migrations/000004_create_chapter_table.up.sql:1]: chapter_table.down.sql is empty?

**Review by LeoHChen — COMMENTED** [review]
I am thinking we shall have a naming prefix for database tables for easier management later on.  Can you propose a naming themes for DB tables?

File: api/migrations/000003_create_nft_table.up.sql (hunk 1 above)
Comment (edisonz0718) [reply api/migrations/000003_create_nft_table.up.sql:10]: Do we want to use decentralized storage for this project? We can easily add a new column if we need it later

File: api/migrations/000004_create_chapter_table.up.sql (hunk 2 above)
Comment (edisonz0718) [reply api/migrations/000004_create_chapter_table.up.sql:1]: good catch. Fixed

**Comment by edisonz0718** [comment]
> I am thinking we shall have a naming prefix for database tables for easier management later on. Can you propose a naming themes for DB tables?

Yes. Propose a scheme here: [DB naming schema](https://www.notion.so/storyprotocol/DB-Naming-schema-638f8f681dd140c493b15a7764ffa0c1)

The scheme:
**wallet_**: Data associates with wallet, i.e.
- wallet_merkle_proof

**nft_**: nft related data, i.e.
- nft_collection
- nft_allowlist
- nft_token

**story_**: content related, i.e.
- story_franchise
- story_info
- story_chapter

Updated the script based on this scheme

File: api/migrations/000003_create_nft_table.up.sql (hunk 1 above)
Comment (LeoHChen) [reply api/migrations/000003_create_nft_table.up.sql:10]: we can add later, please just create a github issue then.

**Review by LeoHChen — APPROVED** [review]

**Pull request merged**